src/dftbp/mixer/factory.F90

```diff
@@ -0,0 +1,72 @@
+!--------------------------------------------------------------------------------------------------!
+!  DFTB+: general package for performing fast atomistic simulations                                !
+!  Copyright (C) 2006 - 2023  DFTB+ developers group                                               !
+!                                                                                                  !
+!  See the LICENSE file for terms of usage and distribution.                                       !
+!--------------------------------------------------------------------------------------------------!
+
+#:include 'common.fypp'
+#:set FLAVOURS = [('real', 'real', 'Real'),('cmplx', 'complex', 'Cmplx')]
+#:set VARIANTS = ['Anderson', 'Broyden', 'Diis', 'Simple']
+module dftbp_mixer_factory
+  use dftbp_io_message, only : error
+
+  use dftbp_mixer_mixer, only : TMixerReal, TMixerCmplx
+  #:for VARIANT in VARIANTS
+  use dftbp_mixer_${VARIANT}$mixer, only : T${VARIANT}$MixerInp
+    #:for NAME, TYPE, LABEL in FLAVOURS
+    use dftbp_mixer_${VARIANT}$mixer, only : T${VARIANT}$Mixer${LABEL}$, T${VARIANT}$Mixer${LABEL}$_init
+    #:endfor
+  #:endfor
+
+  implicit none 
+  private
+  public :: TMixerInput, TMixerFactoryReal, TMixerFactoryCmplx
+
+
+  type :: TMixerInput
+    type(TAndersonMixerInp), allocatable :: andersonMixerInp
+    type(TBroydenMixerInp), allocatable :: broydenMixerInp
+    type(TDiisMixerInp), allocatable :: diisMixerInp
+    type(TSimpleMixerInp), allocatable :: simpleMixerInp
+  end type TMixerInput
+
+
+  contains
```

Comment:
NOte indentation (`contains` is on the same level as the beginning and end of the construct it belongs to. Otherwise indentation is always by 2 chars)